testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -299,6 +312,60 @@ public void testViewColumnAccessControl()
+    {
+        TestingSystemSecurityMetadata systemSecurityMetadata = (TestingSystemSecurityMetadata) getDistributedQueryRunner().getCoordinator().getInstance(Key.get(SystemSecurityMetadata.class));
+        Session viewOwnerSession = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.forUser("test_view_access_owner")
+                        .withEnabledRoles(Set.of("ownerrole"))
+                        .build())
+                .setCatalog(getSession().getCatalog())
+                .setSchema(getSession().getSchema())
+                .build();
+
+        String columnAccessViewName = "test_view_column_access_" + randomNameSuffix();
+
+        systemSecurityMetadata.grantRoles(getSession(), Set.of("ownerrole"), Set.of(new TrinoPrincipal(USER, "test_view_access_owner")), false, Optional.empty());
+        systemSecurityMetadata.setViewOwner(
+                getSession(),
+                new CatalogSchemaTableName("blackhole", "default", columnAccessViewName),
+                new TrinoPrincipal(USER, viewOwnerSession.getUser()));
+
+        assertAccessAllowed(
+                viewOwnerSession,
+                "CREATE VIEW " + columnAccessViewName + " AS SELECT * FROM orders",
+                privilege("orders", CREATE_VIEW_WITH_SELECT_COLUMNS));
+
+        // verify that role permissions are taken for the view owner's role grants
+        // even if executed by the owner with different roles enabled
+        executeExclusively(() -> {
+            systemSecurityMetadata.grantRoles(getSession(), Set.of("deniedrole"), Set.of(new TrinoPrincipal(USER, "test_view_access_owner")), false, Optional.empty());
+            try {
+                getQueryRunner().getAccessControl().denyIdentityTable((identity, table) -> !identity.getEnabledRoles().contains("deniedrole") || !"orders".equals(table));
+                // TestingAccessControlManager works by denying accesses, so:
+                // * viewOwnerSession does not have deniedrole enabled
+                // * the owner has a role grant to deniedrole
+                // we deny access to deniedrole, so:
+                // * even though viewOwnerSession would be allowed,
+                // * the session created for view owner will be denied
+                assertThatThrownBy(() -> getQueryRunner().execute(viewOwnerSession, "SELECT * FROM " + columnAccessViewName))
+                        .hasMessageMatching("Access Denied: Cannot select from columns \\[.*] in table or view \\w+\\.\\w+\\.orders");
+
+                // verify view can be queried when the role grant is revoked
+                systemSecurityMetadata.revokeRoles(getSession(), Set.of("deniedrole"), Set.of(new TrinoPrincipal(USER, "test_view_access_owner")), false, Optional.empty());
+                getQueryRunner().execute(viewOwnerSession, "SELECT * FROM " + columnAccessViewName);
+            }
+            finally {
+                getQueryRunner().getAccessControl().reset();
```

Comment:
In this class, this is typicaly done by `assertDenied` or similar, so that test methods don't need `try-finally`